### ISSUE TITLE
Respect Inline Suggestion to improve compatibility with GitHub Copilot

### ISSUE THREAD

**Issue by adam-coster** [yzhang-gh/vscode-markdown#1011]
**Issue:** The inline suggestions provided by GitHub Copilot cannot be used in Markdown files while Markdown All In One is active.

**Feature Request:** Can we get a config option to toggle allowing <kbd>Tab</kbd> to take inline suggestions?

**Update** (@yzhang-gh): It has been fixed in the [dev build](https://github.com/yzhang-gh/vscode-markdown#latest-development-build).

## Background & Workaround

I changed the `markdown.extension.onTabKey` keyboard shortcut to add `!inlineSuggestionVisible` to the `when` conditions. This disables the Markdown extension's <kbd>Tab</kbd> behavior when Copilot is showing a suggestion, allowing the  `editor.action.inlineSuggest.commit` binding on <kbd>Tab</kbd> to take over.

Whether or not this option gets added, for anyone else needing to enable Copilot or other inline-suggestion tools to work when using this extension, you can update your `keybindings.json` (<kbd><kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>p</kbd></kbd>, then find `Preferences: Open Keyboard Shortcuts (JSON)`) by adding the following:

```json
[
  {
    "key": "tab",
    "command": "markdown.extension.onTabKey",
    "when": "editorTextFocus && !inlineSuggestionVisible && !editorReadonly && !editorTabMovesFocus && !hasOtherSuggestions && !hasSnippetCompletions && !inSnippetMode && !suggestWidgetVisible && editorLangId == 'markdown'"
  },
  {
    "key": "tab",
    "command": "-markdown.extension.onTabKey",
    "when": "editorTextFocus && !editorReadonly && !editorTabMovesFocus && !hasOtherSuggestions && !hasSnippetCompletions && !inSnippetMode && !suggestWidgetVisible && editorLangId == 'markdown'"
  }
]
```

The first entry adds the `!inlineSuggestionVisible` condition, and the second entry essentially blocks the shortcut that comes with the Markdown All In One extension.

**Comment by adam-coster** [comment]
Additional reference from the VSCode repo: microsoft/vscode#131953

**Comment by Lemmingh** [comment]
Similar to #665

I wonder if there is some relation.

**Comment by adam-coster** [comment]
Yep, same issue!

**Comment by yzhang-gh** [comment]
We can add the `!inlineSuggestionVisible` but I think we also need to ask VS Code to document it (what is its meaning and when is it used).

**Comment by MitchTalmadge** [comment]
I've confirmed this works to solve the problem I was experiencing in https://github.com/microsoft/vscode/issues/131953. Thank you for this solution.

**Comment by JoyShaheb** [comment]
The Solution provided by @adam-coster worked for me, Thank you !

**Comment by Djcarrillo6** [comment]
Solution referenced above works for me as well! Thank you!!

**Comment by fonnesbeck** [comment]
This does not work for me on Windows (nor does uninstalling Markdown All In One).

**Comment by Franky1** [comment]
I tried the solution, but could not edit the `keybindings.json` file. VSCode says it is write protected. I found no option to unlock.
- OS Windows 10
- VSCode Version `1.63.2`

**Comment by yzhang-gh** [comment]
Just a note, you can install the [dev build](https://github.com/yzhang-gh/vscode-markdown#latest-development-build) to get the fix.

**Comment by sasukeh** [comment]
@adam-coster 's workaround is works for me for now! Thanks  @adam-coster .

**Comment by laakal** [comment]
It's also conflicts with the [docs-markdown](https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-markdown) extension. It works after deleting the docs-markdown extension.

**Comment by anbarasanv** [comment]
Thank you very much works fine for me on my Mac.

**Comment by yzhang-gh** [comment]
has been fixed in v3.4.2

**Comment by MitchTalmadge** [comment]
Thank you so much
